### PR TITLE
block job with install: Add one keyword to detect install start

### PR DESCRIPTION
**Pull request by zhencliu**
blockdev_commit_install/blockdev_mirror_install
  blockdev_snapshot_install/blockdev_stream_install
  updated, please be noted that we have to add more
  keywords if the installation console logs changed

Signed-off-by: Zhenchao Liu <zhencliu@redhat.com>

ID: 1888498

### CODE REVIEW

**Comment by zhencliu** [comment]
(1/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_stream_install.q35: PASS (1490.33 s)
 (2/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_snapshot_install.q35: PASS (1632.72 s)
 (3/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_mirror_install.q35: PASS (1783.29 s)
 (4/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_commit_install.q35: PASS (1365.78 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-10-21T07.04-98a9bad/results.html
JOB TIME   : 6278.49 s

hi @YongxueHong @ngu-niny , can you please help to review the patch? Thanks.

File: qemu/tests/blockdev_mirror_install.py [hunk 1]
```diff
@@ -19,7 +20,7 @@ def _is_install_started(self, start_msg):
         out = self.main_vm.serial_console.get_output() \
             if self.main_vm.serial_console else None
         out = '' if out is None else out
-        return start_msg in out
+        return re.search(start_msg, out, re.M) is not None
```
Comment (YongxueHong) [inline qemu/tests/blockdev_mirror_install.py:23]:
```suggestion
        return bool(re.search(start_msg, out, re.M))
```

File: qemu/tests/blockdev_snapshot_install.py [hunk 2]
```diff
@@ -24,8 +25,8 @@ def run(test, params, env):
     def tag_for_install(vm, tag):
         if vm.serial_console:
             serial_output = vm.serial_console.get_output()
-            if serial_output and tag in serial_output:
-                return True
+            if serial_output:
+                return re.search(tag, serial_output, re.M) is not None
```
Comment (YongxueHong) [inline qemu/tests/blockdev_snapshot_install.py:29]:
```suggestion
            if serial_output and re.search(tag, serial_output, re.M):
                return True
```
whatever the boolean value of `re.search(tag, serial_output, re.M) is not None ` is False or True, the next `logging.info("VM has not started yet")` will not be called.

File: qemu/tests/blockdev_mirror_install.py (hunk 1 above)
Comment (zhencliu) [reply qemu/tests/blockdev_mirror_install.py]: updated.

File: qemu/tests/blockdev_snapshot_install.py (hunk 2 above)
Comment (zhencliu) [reply qemu/tests/blockdev_snapshot_install.py]: updated

**Review by YongxueHong — APPROVED** [review]

**Comment by zhencliu** [comment]
(1/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_stream_install.q35: PASS (1517.44 s)
 (2/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_snapshot_install.q35: PASS (1376.44 s)
 (3/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_mirror_install.q35: PASS (1670.56 s)
 (4/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_commit_install.q35: PASS (1602.13 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-10-21T10.01-ee0c5cb/results.html
JOB TIME   : 6173.11 s

**Comment by ngu-niny** [comment]
LGTM and test PASS:

 (1/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.ppc64le.io-github-autotest-qemu.blockdev_commit_install:  PASS (1150.46 s)
 (2/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.ppc64le.io-github-autotest-qemu.blockdev_stream_install:  PASS (1183.10 s)
 (3/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.ppc64le.io-github-autotest-qemu.blockdev_snapshot_install:  PASS (1097.48 s)
 (4/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.ppc64le.io-github-autotest-qemu.blockdev_mirror_install:  PASS (1342.11 s)

Acked-by: Nini Gu <ngu@redhat.com>

**Pull request merged**